packages/server-common/test/handlers/user/invite.handler.test.ts

```diff
@@ -1,174 +1,134 @@
 
-    const event = getEventBridgeEventMock(restUserMock().id);
+    const event = getEventBridgeEventMock(userMock().id);
 
     await expect(inviteHandler(event)).rejects.toThrow(
-      `Unable to save the code for the user with ID ${restUserMock().id}`,
+      `Unable to save the code for the user with ID ${userMock().id}`,
     );
   });
 
   test('Should throw when it fails to send the email but still save the new invitation code', async () => {
-    const userWithoutConnection: RestUser = {
-      ...restUserMock(),
-      data: {
-        ...restUserMock().data,
-        connections: {
-          iv: [],
-        },
-      },
+    const userWithoutConnection: UserDataObject = {
+      ...userMock(),
+      connections: [],
     };
-    userClient.fetchById.mockResolvedValueOnce(userWithoutConnection);
+    dataProvider.fetchById.mockResolvedValueOnce(userWithoutConnection);
+    dataProvider.update.mockResolvedValueOnce(null);
     sendEmailMock.mockRejectedValueOnce(new Error('some error'));
 
-    const event = getEventBridgeEventMock(restUserMock().id);
+    const event = getEventBridgeEventMock(userMock().id);
 
     await expect(inviteHandler(event)).rejects.toThrow(
-      `Unable to send the email for the user with ID ${restUserMock().id}`,
+      `Unable to send the email for the user with ID ${userMock().id}`,
     );
-    expect(userClient.patch).toBeCalledWith(userWithoutConnection.id, {
-      connections: {
-        iv: [{ code: expect.any(String) }],
-      },
+    expect(dataProvider.update).toBeCalledWith(userWithoutConnection.id, {
+      connections: [{ code: expect.any(String) }],
     });
   });
 
   test('Should not send the invitation email for a user that already has the invitation code', async () => {
     const code = 'c6fdb21b-32f3-4549-ac17-d0c83dc5335b';
-    const userWithConnection: RestUser = {
-      ...restUserMock(),
-      data: {
-        ...restUserMock().data,
-        connections: {
-          iv: [
-            {
-              code,
-            },
-          ],
-        },
-      },
+    const userWithConnection: UserDataObject = {
+      ...userMock(),
+      connections: [{ code }],
     };
-    userClient.fetchById.mockResolvedValueOnce(userWithConnection);
+    dataProvider.fetchById.mockResolvedValueOnce(userWithConnection);
 
-    const event = getEventBridgeEventMock(restUserMock().id);
+    const event = getEventBridgeEventMock(userMock().id);
 
     await inviteHandler(event);
 
-    expect(userClient.fetchById).toBeCalledWith(userWithConnection.id);
+    expect(dataProvider.fetchById).toBeCalledWith(userWithConnection.id);
     expect(sendEmailMock).not.toBeCalled();
   });
 
-  test('Should find the user with an existing connection code and add an invitation code without removing the existing connection', async () => {
+  test('Should not send the invitation email for a user that already has an authentication code', async () => {
```

Comment:
you have repurposed a test that was referring to a whole different part of the app which I don't think you should have changed (see my comment below) - unless I am mistaken?

If I am right - I think this should be an new test

Reply:
The removed test and the new test are mutually exclusive - the old test asserts that the invitation code is added to a user with a pre-existing auth code, and the new test asserts that it is _not_.